### ISSUE TITLE
[Bug] [$2000] The browser tab shows (1) New Expensify even though there are no unread messages in the LHN

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#8005]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open app in Web 
2. Check browser tab 

## Expected Result:
When user has no unread messages it should not display any counter. 

## Actual Result:
The browser tab shows (1) New Expensify even though there are no unread messages in the LHN

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Web

**Version Number:** 1.1.41-0
**Reproducible in staging?:**  Y 
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 

![Screen Shot 2022-03-04 at 12 06 18 AM](https://user-images.githubusercontent.com/43995119/156703305-30f32d05-a330-4f5d-ad08-f5a985b0121b.png)

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @quinthar 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1646285973562329

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @NikkiWines (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by NikkiWines** [comment]
Made a snippet to show the keyvaluepairs containing an `unreadActionCount` greater than 0 
```JS
let connection = indexedDB.open('OnyxDB', 2);
let unreadReportKVPs;

connection.onsuccess = (e) => {
    const database = e.target.result;
    const transaction = database.transaction(['keyvaluepairs']);
    const objectStore = transaction.objectStore('keyvaluepairs');
    const keyvaluepairsPromise = objectStore.getAll();

    keyvaluepairsPromise.onsuccess = (e) => {
        let keyvaluepairs = e.target.result;
        unreadReportKVPs = keyvaluepairs.filter((object) => {
            if (typeof(object) === 'object') {
                return object.unreadActionCount > 0;
            }
            return false;
        });
        console.log(unreadReportKVPs);
    };
};
```

**Comment by NikkiWines** [comment]
I think this can probably be an external issue? I'm not personally experiencing this so it's a bit difficult to debug.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @adelekennedy (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by adelekennedy** [comment]
[Internal](https://www.upwork.com/ab/applicants/1499900368198955008/job-details)
[External](https://www.upwork.com/jobs/~01b20e0835268d79ec)

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @Santhosh-Sellavel (`Exported`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @Beamanator (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by adelekennedy** [comment]
@Santhosh-Sellavel @Beamanator it looks like the slack convo is [still evolving](https://expensify.slack.com/archives/C01GTK53T8Q/p1646285973562329) but I wanted to get the job posted this week

**Comment by K4tsuki** [comment]
#### proposal

The problem is there is (are)  report(s) that has unread message but the report is not displayed in LHN.
LHN doesn't display all reports retrieved from `Get chatList,` it filters out some reports.

https://github.com/Expensify/App/blob/d4ed740be06862eb4ea31fa831de9416f7ed0a55/src/libs/OptionsListUtils.js#L740

<br />

My solution is to count unread messages only in reports that are  displayed in LHN.

**Comment by Santhosh-Sellavel** [comment]
> The problem is there is (are) report(s) that has unread message but the report is not displayed in LHN.
LHN doesn't display all reports retrieved from Get chatList, it filters out some reports.

@K4tsuki Can you answer the following questions.
1. What reports are filtered out? 
2. Why its get filtered?

**Comment by K4tsuki** [comment]
@Santhosh-Sellavel I have updated the proposal. The problem is likely from -1 inside `unreadActionCounts`. In my local reports data there is a report with notification prefecerence set to `daily`, it is a policyRoom.

**Comment by Santhosh-Sellavel** [comment]
> The problem is likely from -1 inside unreadActionCounts. In my local reports data there is a report with notification prefecerence set to daily, it is a policyRoom.

@K4tsuki Can you clearly make a new proposal, with the changes you are proposing.

**Comment by K4tsuki** [comment]
#### proposal

The title say: `The browser tab shows (1) New Expensify even though there are no unread messages in the LHN`

So I assume the problem only happens if there is no unread message and if there is a message, the count badge  show correctly.

The problem is because there is an unread message in the room with notification preference set to `daily`.

The daily notification preference is set in here:

https://github.com/Expensify/App/blob/d4ed740be06862eb4ea31fa831de9416f7ed0a55/src/libs/actions/Report.js#L193-L195

If the room is `ReportUtils.isChatRoom`, that is  `isUserCreatedPolicyRoom(report) || isDefaultRoom(report);` then default notification preferences is set to `daily`.

If there is an unread message in the room: the `unreadActiionCount` of that room for badge updater will be -1 set by:

https://github.com/Expensify/App/blob/d4ed740be06862eb4ea31fa831de9416f7ed0a55/src/libs/UnreadIndicatorUpdater/index.js#L44-L48

Then, In final functions to show `unreadCount`

https://github.com/Expensify/App/blob/d4ed740be06862eb4ea31fa831de9416f7ed0a55/src/libs/UnreadIndicatorUpdater/index.js#L15-L21

If total count is 0 and there is a `-1` unreadActionCount in a report, then the final unreadCount badge would be `1`.

So I assume it is only happens for developer, because `isUserCreatedPolicyRoom || isDefaultRoom` aren't displayed in production right?

<br />

So my solution same as above, my solution is to count unread messages only in reports that are displayed in LHN.
Maybe make an Onyx key LHN_chat list or lib utils to store current LHN chat list and use it in:

https://github.com/Expensify/App/blob/d4ed740be06862eb4ea31fa831de9416f7ed0a55/src/libs/UnreadIndicatorUpdater/index.js#L14-L24

<br />

@Santhosh-Sellavel for your question: 

> 1.  What reports are filtered out?

At least`isUserCreatedPolicyRoom` and `isDefaultRoom` are filtered in production. 
Others filtered rooms are based on the logic in:

https://github.com/Expensify/App/blob/d4ed740be06862eb4ea31fa831de9416f7ed0a55/src/libs/OptionsListUtils.js#L440-L442

and

https://github.com/Expensify/App/blob/d4ed740be06862eb4ea31fa831de9416f7ed0a55/src/libs/OptionsListUtils.js#L453-L471


> 2. Why its get filtered?

For `isUserCreatedPolicyRoom` and `isDefaultRoom` I think the rooms are still in beta. For others rooms, there are many criteria.

<br />

@Santhosh-Sellavel  What do you think? are there something missing  with my proposal?

**Comment by Beamanator** [comment]
@Santhosh-Sellavel to be clear, are you thinking this issue will be resolved once the room features are out of beta? If yes can you explain why?

Also @K4tsuki thanks so much for your ideas so far, are you able to reproduce the issue regularly? If so, I assume you've tested your proposal with the reproduction steps and you're not seeing the issue anymore with your proposed changes?

**Comment by K4tsuki** [comment]
@Beamanator 

>  are you able to reproduce the issue regularly?

Yes, I can reproduce it regularly. Even before the issue is posted I already have this issue (I am assuming the source of my bug is same as this issue). 

> you've tested your proposal with the reproduction steps and you're not seeing the issue anymore with your proposed changes

I have tested it and it solves this issue, here is the video:


https://user-images.githubusercontent.com/1313124/158597146-a4c70028-a3d7-4fcd-8956-8b93c4a821d0.mp4

Left tab is production release new expensify and right tab  is my localhost development build.

**Comment by Santhosh-Sellavel** [comment]
>@Santhosh-Sellavel to be clear, are you thinking this issue will be resolved once the room features are out of beta? If yes can you explain why?
Yes, I should have explained why, sorry I missed out earlier.

The issue occurs due to messages received in default or user-created rooms. Look at the attachments below with/without beta for the room feature. Notification count in the header doesn’t make sense only when the user does not have the beta. So I believe when the room features are out of beta, this issue becomes invalid.

#### If room beta is available
<img width="788" alt="Screenshot 2022-03-16 at 7 42 01 PM" src="https://user-images.githubusercontent.com/85645967/158609343-52ab68d1-1247-4f0f-a1a1-63fbbee0ac20.png">

#### if room beta is not available
<img width="876" alt="Screenshot 2022-03-16 at 7 32 50 PM" src="https://user-images.githubusercontent.com/85645967/158607878-11e8459a-541a-439a-b541-8fd9f171e850.png">

cc: @Beamanator

**Comment by Santhosh-Sellavel** [comment]
@Beamanator 

The idea of creating ONYXKey for LHN is a bad idea, so big NO!

I am interested in listening more about this idea.
<img width="821" alt="Screenshot 2022-03-16 at 8 22 44 PM" src="https://user-images.githubusercontent.com/85645967/158619070-a79e1d12-b009-46c3-8c17-907a0942206a.png">

@K4tsuki could you elaborate more about this one?

**Comment by K4tsuki** [comment]
@Santhosh-Sellavel 
We could add

```
if (ReportUtils.isChatRoom(report.reportID) && (!Permissions.canUseDefaultRooms(betas))) { 
    return; 
}
```

Below:

https://github.com/Expensify/App/blob/7919e24c808dea9dedfe4a2d513365cf092cc1e1/src/libs/UnreadIndicatorUpdater/index.js#L40-L42

<br />

It filters out default rooms for users who cannot use default rooms yet because of beta.

**Comment by Santhosh-Sellavel** [comment]
@Beamanator

**Comment by Beamanator** [comment]
Let's put this idea on hold for a little bit since we're discussing a few things internally - we'll get back to reviewing proposals soon!

**Comment by adelekennedy** [comment]
silly question @Beamanator anything else I need to do to put this on hold besides add HOLD to the title?

**Comment by Beamanator** [comment]
Good question @adelekennedy , that's all we need to do! :D

**Comment by Beamanator** [comment]
Ok taking off hold & adding the reward to the start of the title - @adelekennedy I think it may be time to double the price of this one?

**Comment by adelekennedy** [comment]
@Beamanator yep - I got the Slack ping right before we put it on hold! I'll double now!

**Comment by thesahindia** [comment]
### Proposal 
This happens when chat room has unread message but as it's behind the beta so the chat doesn't show up at the LHN.
We can just use the code given below so the the tab won't show (1) unread message
```js
            if (ReportUtils.isChatRoom(report.reportID) && (!Permissions.canUseDefaultRooms(betas))) {
                return;
            }
```

we just need to add it after -
https://github.com/Expensify/App/blob/4d646e79cc006aefe4c530b19cb085e1107c9042/src/libs/UnreadIndicatorUpdater/index.js#L40-L43

**Comment by thesahindia** [comment]
What I found is that we can filter out user created rooms to fix this( like proposed above ) but there's only one problem if you create a new room and send a message then the unread count will show up but it will go away on refreshing the page.
Other than that it works fine.

**Comment by Santhosh-Sellavel** [comment]
@thesahindia 
Is this your final proposal [here](https://github.com/Expensify/App/issues/8005#issuecomment-1078418150)
Is there any code to add/remove?

**Comment by thesahindia** [comment]
Yeah we have to get betas and for that we can use `onyx.connect` 
and forgot to change one more thing instead of `!Permissions.canUseDefaultRooms(betas)` we need to use `!Permissions.canUsePolicyRooms(betas)`

### Final Proposal 
```diff
let connectionID;
+let betas;

/**
 * Bind to the report collection key and update
 * the title and unread count indicators
 */

function listenForReportChanges() {
+    betas = Onyx.connect({
+        key: ONYXKEYS.BETAS,
+        callback: val => betas = val,
+    });
    connectionID = Onyx.connect({
        key: ONYXKEYS.COLLECTION.REPORT,

        callback: (report) => {
            if (!report || !report.reportID) {
                return;
            }

+            if (ReportUtils.isUserCreatedPolicyRoom(report) && (!Permissions.canUsePolicyRooms(betas))) {
+                return;
+            }
```

**Comment by Santhosh-Sellavel** [comment]
@thesahindia 
This does not make any sense to me, what are you trying to do here?
```
 betas = Onyx.connect({
+        key: ONYXKEYS.BETAS,
+        callback: val => betas = val,
+    });
```
We should also include check for Default rooms too, because they are in beta too.
Even after adding those checks the issue persisted for me.

**Comment by Santhosh-Sellavel** [comment]
> Even after adding those checks the issue persisted for me.
Case:
Where all `users` has access to both` user room` &` default room` betas.
User create a room & but didn’t send any messages. Also by default notification priority is `Daily` for the created rooms for all users.

Show daily even there is no messages we still some how show (1) notification in the browser tab for those rooms with no messages.

Can we handle this seperatly or what can we do here?
@Beamanator

**Comment by thesahindia** [comment]
Ah my bad I was testing something and didn't change the code 😬
Updated the proposal can you check now.
>We should also include check for Default rooms too, because they are in beta too.

We show the default rooms in LHN so I think we shouldn't remove the unread message count for them

**Comment by Santhosh-Sellavel** [comment]
> We show the default rooms in LHN so I think we shouldn't remove the unread message count for them

No, we should add check for defaultRooms too. Because see images below,
### In dev with default room beta
<img width="564" alt="Screenshot 2022-03-28 at 11 30 08 PM" src="https://user-images.githubusercontent.com/85645967/160459122-a78a784e-6466-489e-b1a7-c3947e9459bb.png">

### In staging without default room beta
<img width="405" alt="Screenshot 2022-03-28 at 11 30 30 PM" src="https://user-images.githubusercontent.com/85645967/160459145-93d6e106-56f4-4644-ada9-cf8996b0a20d.png">

**Comment by thesahindia** [comment]
For that we can add this one-
```js
            if (ReportUtils.isDefaultRoom(report) && (!Permissions.canUseDefaultRooms(betas))) {
                return;
            }
```

**Comment by Santhosh-Sellavel** [comment]
Waiting for @Beamanator comments on this https://github.com/Expensify/App/issues/8005#issuecomment-1080854713

**Comment by Beamanator** [comment]
@Santhosh-Sellavel 

Can you please clarify the two (or more) different cases you're mentioning and why you think one case should be handled separately?

**Comment by Santhosh-Sellavel** [comment]
@Beamanator 
Case 1: This issue in which the user is shown notifications count for chats they can’t access because `they don't have access to the betas.`

[Case 2](https://github.com/Expensify/App/issues/8005#issuecomment-1080854713). `The User has access to the beta`, notifications count is shown for the user. But there are no messages in the room So it's not displayed in LHN. Without opening the chat notification count will not go away. I felt this is like a separate issue. 

What do you say?

**Comment by Beamanator** [comment]
Ok great thanks for explaining both cases @Santhosh-Sellavel ! I think both cases are similar enough and both need to be solved to truly resolve the "browser tab shows (1) New Expensify even though there are no unread messages in the LHN" issue. For that reason I think it's fair to solve both cases in this issue.

@mallenexpensify thanks for the tip! 👍

**Comment by Santhosh-Sellavel** [comment]
Thanks, @Beamanator.
So whats the expectation for case 2?

**Comment by Beamanator** [comment]
Exactly - if the room isn't shown in the LHN we shouldn't show its unread messages, or else (as you said) they won't be able to be cleared which would be very unpleasant

**Comment by Santhosh-Sellavel** [comment]
Open to proposals again.
@thesahindia update your proposal to handle both the cases thanks!

**Comment by adelekennedy** [comment]
any updates here? @thesahindia do you have an updated proposal the team can review?

**Comment by thesahindia** [comment]
Nope I couldn't update my proposal because I am getting error when trying to create a room, I have reported it on [slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1648760293684219)

**Comment by adelekennedy** [comment]
Just got the ping to double this again!

**Comment by adelekennedy** [comment]
Doubling this issue again, @thesahindia it sounds like your proposal above will work once the bug is fixed for creating rooms?

**Comment by thesahindia** [comment]
Yeah it will solve the issue but according to https://github.com/Expensify/App/issues/8005#issuecomment-1081840317 there's another issue but I can't reproduce it.

**Comment by Santhosh-Sellavel** [comment]
Let's not double this further @adelekennedy, its already overpriced. cc: @Beamanator

**Comment by Beamanator** [comment]
@Santhosh-Sellavel yes it may be a high reward, but since we still don't have an acceptable proposal I think it's fine to keep doubling till we do 👍

**Comment by adelekennedy** [comment]
I just got the doubling ping again - @Beamanator and @Santhosh-Sellavel it sounds like we're moving forward with doubling until we get a proposal but is there anything else we need to address here to get this issue worked on? It seems like @thesahindia proposal will work but we're waiting on this [linked issue](https://github.com/Expensify/App/issues/8005#issuecomment-1081840317) being solved?

**Comment by Santhosh-Sellavel** [comment]
Room creation has some errors, I am also not able to reproduce the [Case 2](https://github.com/Expensify/App/issues/8005#issuecomment-1081840317)

Let's wait until this issue https://github.com/Expensify/App/issues/8534 gets solved, or I'll try reproducing this one again and confirm by end of this week.

cc: @Beamanator @adelekennedy 

Also one help @Beamanator, can you give a try reproducing this issue.
> The User has access to the beta, notifications count is shown for the user. But there are no messages in the room So it's not displayed in LHN. Without opening the chat notification count will not go away. I felt this is like a separate issue.

**Comment by Beamanator** [comment]
Sorry for the delay @Santhosh-Sellavel I'll try to help reproduce tomorrow 👍

**Comment by adelekennedy** [comment]
We're at the doubling point again but it sounds like we're still holding on reproducing the issue - @Beamanator @Santhosh-Sellavel

**Comment by Santhosh-Sellavel** [comment]
@adelekennedy 
We are almost there. 
Let's hold for couple more days wait for @Beamanator reply and proceed further.

Thanks

**Comment by Beamanator** [comment]
Yeah lets not double this one quite yet.

It looks like the PR we were waiting for has been merged: https://github.com/Expensify/App/issues/8534 sorry this week has been pretty busy with internal priorities. @Santhosh-Sellavel did you try to reproduce last week? I will have time again Monday or Tuesday next week to help trying to reproduce if you haven't by then

**Comment by Santhosh-Sellavel** [comment]
I'm unable to reproduce this one, [Case 2](https://github.com/Expensify/App/issues/8005#issuecomment-1081840317)

**Comment by Beamanator** [comment]
@Santhosh-Sellavel I also just tried reproducing Case 2 and didn't see the issue -> I created a room and didn't see the unread count increment at all. I have all betas enabled in my dev environment. I agree let's focus on just addressing Case 1 in this issue 👍

**Comment by Santhosh-Sellavel** [comment]
Thanks, @Beamanator!

@thesahindia Can you regroup & submit your proposal?

**Comment by Santhosh-Sellavel** [comment]
But I think there might even simpler solution!

**Comment by thesahindia** [comment]
### Proposal 

I am suggesting that we should just add two checks, 1st for default rooms and 2nd for user created rooms. 

Code changes -

```diff
+let betas;
+Onyx.connect({
+    key: ONYXKEYS.BETAS,
+    callback: val => betas = val,
+});

function listenForReportChanges() {
    connectionID = Onyx.connect({
        key: ONYXKEYS.COLLECTION.REPORT,
        callback: (report) => {
            if (!report || !report.reportID) {
                return;
            }

            if (report.notificationPreference === CONST.REPORT.NOTIFICATION_PREFERENCE.MUTE) {
                return;
            }

+            if (ReportUtils.isDefaultRoom(report) && (!Permissions.canUseDefaultRooms(betas))) {
+                return;
+            }

+            if (ReportUtils.isUserCreatedPolicyRoom(report) && (!Permissions.canUsePolicyRooms(betas))) {
+                return;
+            }

```

Need to made this changes in [here](https://github.com/Expensify/App/blob/ec828a22f031f646b19af819a27c70512a881712/src/libs/UnreadIndicatorUpdater/index.js#L32)

**Comment by Santhosh-Sellavel** [comment]
The proposal is simple & solves the issue. But it's not synced with LHN in any way.  If there could be util function to find out whether the report should be shown in LHN or not. Which should be consumed by both LHN window & UnreadIndicatorUpdater so the results are always correct.

Any thoughts? @Beamanator

**Comment by Beamanator** [comment]
@Santhosh-Sellavel that does sound like a good idea, if we're planning to **hide** Default room / Policy rooms in the LHN when there's no messages. I don't see that happening in `SidebarLinks.js`, do you see if that's happening somewhere?

**Comment by Beamanator** [comment]
Hmm it looks like in that code we're only hiding the rooms if there's no participants in the room - `_.isEmpty(logins)` right?

And in that case we hopefully wouldn't count unread messages since the user also wouldn't be part of that room

**Comment by Santhosh-Sellavel** [comment]
> @Santhosh-Sellavel Hmm it looks like in that code we're only hiding the rooms if there's no participants in the room - _.isEmpty(logins) right?

No please check the whole snippet, that's one of the filters!

**Comment by thesahindia** [comment]
If we want a util function we can create one like this -
```js
function filterOutBetaRooms(report, betas, excludeDefaultRooms = false) {
    if (ReportUtils.isChatRoom(report) && (!Permissions.canUseDefaultRooms(betas) || excludeDefaultRooms)) {
        return false;
    }

    if (ReportUtils.isPolicyExpenseChat(report) && !Permissions.canUsePolicyExpenseChat(betas)) {
        return false;
    }

    if (ReportUtils.isUserCreatedPolicyRoom(report) && !Permissions.canUsePolicyRooms(betas)) {
        return false;
    }
    return true;
}
```

and we can add a check like this -
```js
  if (!filterOutBetaRooms((report, betas, excludeDefaultRooms)) {
            return;
        }
```

instead of this:
https://github.com/Expensify/App/blob/3cdd9a2ed7560140e2b82b5c63b58e709efc3f9a/src/libs/OptionsListUtils.js#L441-L449

Note: We used to show default rooms on searching a user if he is also inside the default room so to fix that a PR introduced `excludeDefaultRoom` flag, however tried removing this flag and the results were fine so I think maybe we don't need it anymore?

We can use the same function [here](https://github.com/Expensify/App/blob/ec828a22f031f646b19af819a27c70512a881712/src/libs/UnreadIndicatorUpdater/index.js#L32)  -

```diff
+let betas;
+Onyx.connect({
+    key: ONYXKEYS.BETAS,
+    callback: val => betas = val,
+});

function listenForReportChanges() {
    connectionID = Onyx.connect({
        key: ONYXKEYS.COLLECTION.REPORT,
        callback: (report) => {
            if (!report || !report.reportID) {
                return;
            }

+           if (!OptionListUtils.filterOutBetaRooms(report, betas)) {
+                return;
+            }

```

**Comment by Beamanator** [comment]
@Santhosh-Sellavel it looks like you're talking about https://github.com/Expensify/App/blob/a2f8cf96d095ded70a15d04ee69bd37be2638c14/src/libs/OptionsListUtils.js#L432-L440 right? I didn't see that the code you linked was 58 lines since the preview only shows 11 lines 😅

**Comment by Beamanator** [comment]
@Santhosh-Sellavel can you please review the proposal ^ ?

**Comment by Santhosh-Sellavel** [comment]
@Beamanator 
The [proposal](https://github.com/Expensify/App/issues/8005#issuecomment-1124900679) mostly looks good. i.e simple enough to handle the beta checks. Just worried about non-sync with LHN. Since there are few other filtering conditions to consider in the report for options.

On second thought I feel these beta checks should be enough for now, for a fail-safe we could use the below check too
for the case where we somehow saw notification count for a room with no messages.
https://github.com/Expensify/App/blob/a2f8cf96d095ded70a15d04ee69bd37be2638c14/src/libs/OptionsListUtils.js#L425

If you are okay with this then let's hire @thesahindia!

Or just wait for a better proposal to address with LHN Synced logic.

**Comment by Beamanator** [comment]
I think the proposal looks great for now, let's move forward and make sure we test thoroughly 👍

**Comment by adelekennedy** [comment]
hired!

**Comment by marcaaron** [comment]
Anything related to unreads should be on hold.

**Comment by Beamanator** [comment]
Oh shoot good point!! Sorry @thesahindia let's put this on hold for now, please don't submit a PR yet 🙏

**Comment by adelekennedy** [comment]
oh hold, not overdue!

**Comment by adelekennedy** [comment]
On hold

**Comment by iwiznia** [comment]
I know this is on hold, but the bug is VERY annoying

**Comment by adelekennedy** [comment]
still holding

**Comment by Beamanator** [comment]
Agreed it's annoying, luckily the Unread Indicator Improvements doc is now out! (thanks to Marc)

**Comment by adelekennedy** [comment]
still on hold (despite how annoying it is 😂 )

**Comment by marcaaron** [comment]
Ok so back with an update here and I think this one can come off HOLD.

We have changed how this feature works and squashed most of the bugs related to Unread Indicators - but this one is a bit elusive as it seems related to be a browser bug. It's not reproducible on Safari AFAICT - yet it is **very consistently reproducible** on Chrome. All you have to do is...

1. Have an unread chat
2. Navigate to the report (see the count clear from the title)
3. Navigate back to the LHN (see the count return)

What's weird is that if you log out the "count" when we programmatically set it then shows `0`, but the title still reverts back to `(1)`.

Somewhat recent SO post about this here: https://stackoverflow.com/questions/72982365/setting-document-title-doesnt-change-the-tabs-text-after-pressing-back-in-the

So it seems related to how Chrome is handling the `document.title`. There is a workaround mentioned in the stack overflow which is basically to do this:

```js
    document.title = '';
    document.title = hasUnread ? `(${totalCount}) ${CONFIG.SITE_TITLE}` : CONFIG.SITE_TITLE;
```

I tried it and it worked (kind of) the count will clear but navigating back will make the old count flash briefly. IMO this is better than telling someone that they have an unread message when they do not, but not totally ideal.

Haven't found much info on this bug so I suspect it is relatively new, but maybe it will just go away with time (or if we complain to the Chome team).

**Comment by marcaaron** [comment]
I'm going to open a PR with the change since I think it is super annoying to see the incorrect count and would rather have a brief incorrect flash than totally incorrect.

**Comment by adelekennedy** [comment]
Are we opening this back up for any proposals? @Beamanator @Santhosh-Sellavel

**Comment by Beamanator** [comment]
Hmm are we able to consistently reproduce anymore? I don't have the `(1)` always there where it used to be, so let's figure out if it's still a problem & occurring for anyone first?

**Comment by marcaaron** [comment]
Yes, it still happens.

**Comment by marcaaron** [comment]
I have a PR to fix it if you want to look @Beamanator https://github.com/Expensify/App/pull/11160

It's hacky but the issue is a browser bug and I have no better way to work around it.

**Comment by adelekennedy** [comment]
Slack convo on payment [here](https://expensify.slack.com/archives/C02NK2DQWUX/p1667872342460039)

**Comment by adelekennedy** [comment]
as discussed in Slack compensation is due for this job! @Santhosh-Sellavel @thesahindia sent you offers

**Comment by Santhosh-Sellavel** [comment]
Accepted thanks @adelekennedy!

**Comment by xtealer** [comment]
> as discussed in Slack compensation is due for this job! @Santhosh-Sellavel @thesahindia sent you offers

Did you get a solution?

**Comment by whroyal** [comment]
As discussed in Slack compensation is due for this job! @jliexpensify @UpworkBartkoski sent you offers

**Comment by xtealer** [comment]
I propose 2 changes. We must update the file `updateUnread` and we must call this function on back navigation events detected to force the stale document.title reference to update.

updateUnread.js should change Line 16 should change from `document.title = '';` to `history.replaceState(null, '');` so file looks like:

```
/**
 * Web browsers have a tab title and favicon which can be updated to show there are unread comments
 */
import CONFIG from '../../../CONFIG';

/**
 * Set the page title on web
 *
 * @param {Number} totalCount
 */
function updateUnread(totalCount) {
    const hasUnread = totalCount !== 0;

    // There is a Chrome browser bug that causes the title to revert back to the previous when we are navigating back. Setting the title to an empty string
    // seems to improve this issue.
    document.title = '';
    document.title = hasUnread ? `(${totalCount}) ${CONFIG.SITE_TITLE}` : CONFIG.SITE_TITLE;
    document.getElementById('favicon').href = hasUnread ? CONFIG.FAVICON.UNREAD : CONFIG.FAVICON.DEFAULT;
}

export default updateUnread;
```
And we can implement a event listener to detect when user navigates back or forward so we can trigger a update when changes are received by the Onyx listeners. This would be implemented on `src/libs/UnreadIndicatorUpdater/index.js`

**Comment by thesahindia** [comment]
Accepted, thanks!

**Comment by thesahindia** [comment]
@xtealer, this doesn't need a solution anymore. Please check others issues with the help wanted label.

**Comment by thesahindia** [comment]
> Accepted, thanks!

Bump @adelekennedy

**Comment by Santhosh-Sellavel** [comment]
@adelekennedy bump!

**Comment by adelekennedy** [comment]
These closed issues always get me - both have been paid out 2k
